src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Extensions/ViewCssScopePass.cs

```diff
@@ -56,29 +56,28 @@ private void ProcessElement(HtmlContentIntermediateNode node, string cssScope)
             }
         }
 
-        bool IsValidElement(IntermediateToken token)
+        static bool IsValidElement(IntermediateToken token)
         {
             var content = token.Content;
-            var isValidToken = content.StartsWith("<", StringComparison.Ordinal)
-                && !content.StartsWith("</", StringComparison.Ordinal)
-                && !content.StartsWith("<!", StringComparison.Ordinal);
             /// <remarks>
             /// We want to avoid adding the CSS scope to elements that do not appear
             /// within the body element of the document. When this pass executes over the
-            /// nodes, we don't have the ability to store whether we are a descandant of a
+            /// nodes, we don't have the ability to store whether we are a descendant of a
             /// `head` or `body` element so it is not possible to discern whether the tag
             /// is valid this way. Instead, we go for a straight-forward check on the tag
             /// name that we are currently inspecting.
             /// </remarks>
-            var isInvalidTag = content.IndexOf("head", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("meta", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("title", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("link", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("base", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("script", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("style", StringComparison.OrdinalIgnoreCase) >= 0
-                || content.IndexOf("html", StringComparison.OrdinalIgnoreCase) >= 0;
-            return isValidToken && !isInvalidTag;
+            return content.StartsWith("<", StringComparison.Ordinal)
```

Comment:
We could probably improve the perf of this a bit if we switched to a span and compared indicies. Not sure how much it's used though so we probably need to profile to see if it's worth doing. Maybe file an issue to follow up on it?

Reply:
> switched to a span and compared indicies

Not sure what you mean, wouldn't expect normal string operations to be slow here as there are no allocations. Obviously there could be better implementations of this, especially since now it's just a bunch of `Equals` calls I guess a static hash set would perform great. But yes, would have to be measured, I imagine there's many places like this in Razor.